plugin/paste_and_format.py

```diff
@@ -0,0 +1,33 @@
+from .core.registry import LspTextCommand
+from .core.typing import Optional
+from .core.views import first_selection_region
+import sublime
+
+
+class LspPasteAndFormatCommand(LspTextCommand):
```

Comment:
Do I understand it correctly that this command is needed only because `paste` doesn't select the pasted text, but `self.view.replace` does so (if it's on a selection), and the non-empty selection(s) is/are needed for `lsp_format_document_range`?

If that is the case, wouldn't it be somewhat "cleaner" to not intercept the `paste` command, but instead to put the formatting logic into `on_post_text_command`? The selections could be derived from the caret positions after pasting and with the clipboard text. I wonder whether this would be simpler or more complicated. Then it wouldn't be necessary to reinvent/reimplement the `paste` command, and in general I'd be a bit hesitant to replace such an essential command like this one, because it might have some unintended side effects (for example what would happen if another package tries to do the same?)

Reply:
> If that is the case, wouldn't it be somewhat "cleaner" to not intercept the paste command, but instead to put the formatting logic into on_post_text_command? 

Interesting idea. 

> The selections could be derived from the caret positions after pasting and with the clipboard text. I wonder whether this would be simpler or more complicated. 

The code for finding the inserted text [looks like this](https://github.com/sublimelsp/LSP/pull/2311/files#diff-982c110867d9809c978a5597e4da66ce7538b327675a66b8f27214413b7c0ec1R518-R527).

> .... it wouldn't be necessary to reinvent/reimplement the paste command, and in general I'd be a bit hesitant to replace such an essential command like this one

You are right. 
I removed the code that tried to re-implement the paste command.

The worst that can happen now is if we do not find the text, and add it to selection,
in which case the code will not be formatted. 

In my testing this worked well.
If you find an example where this doesn't function, please let me know.

I haven't tested multiple cursors formatting on paste,
because I don't know a server that supports the `textDocument/rangesFormatting` capability.